File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/dto/AvailableUserCoupon.java
package com.eager.questioncloud.core.domain.coupon.dto;

import com.eager.questioncloud.core.domain.coupon.enums.CouponType;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class AvailableUserCoupon {
    private Long id;
    private String title;
    private CouponType couponType;
    private int value;
    private LocalDateTime endAt;
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/coupon/dto/AvailableUserCoupon.kt
package com.eager.questioncloud.core.domain.coupon.dto

import com.eager.questioncloud.core.domain.coupon.enums.CouponType
import java.time.LocalDateTime

class AvailableUserCoupon(
    val id: Long,
    val title: String,
    val couponType: CouponType,
    val value: Int,
    val endAt: LocalDateTime
) {

}
